conans/test/integration/editable/test_editable_layout.py

```diff
@@ -34,3 +35,19 @@ def generate(self):
     c.run("install sub2")
     libdir = os.path.join(c.current_folder, "mybuild", "mylibdir")
     assert f"conanfile.py: PKG LIBDIR {libdir}" in c.out
+
+
+def test_editable_matching_folder_names():
+    # https://github.com/conan-io/conan/issues/14091
+    client = TestClient()
+    client.save({"hello-say/conanfile.py": GenConanfile("say", "0.1"),
+                 "hello/conanfile.py": GenConanfile("hello", "0.1").with_settings("build_type")
+                                                                   .with_require("say/0.1")})
+    client.run("editable add hello-say")
+    client.run("build hello-say")
+    client.run("install hello -g CMakeDeps -s:b os=Linux")
+    data = client.load("hello/say-release-data.cmake")
+    hello_say_folder = os.path.join(client.current_folder, "hello-say").replace("\\", "/")
+    assert f'set(say_PACKAGE_FOLDER_RELEASE "{hello_say_folder}")' in data
+    env = client.load("hello/conanbuildenv-release.sh")
+    assert "$script_folder/deactivate_conanbuildenv-release.sh" in env.replace("\\", "/")
```

Comment:
With my planned PR for these scripts to disappear, this tests would fail. Is there another file you can check for?

Reply:
Not really, better check it as my change was breaking this.
It is fine, when you introduce your change, a few tests will have to be fixed anyway, this will be just one more.